aws/rust-runtime/aws-config/src/default_provider.rs

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Default Provider chains for [`region`](region) and credentials (TODO)
+
+use aws_types::config::Config;
+
+use crate::meta::region::ProvideRegion;
+
+pub mod region {
+    //! Default region provider chain
+
+    use crate::environment::region::EnvironmentVariableRegionProvider;
+    use crate::meta::region::ProvideRegion;
+
+    /// Default Region Provider chain
+    ///
+    /// This provider will load region from environment variables.
+    pub fn default_provider() -> impl ProvideRegion {
+        EnvironmentVariableRegionProvider::new()
+    }
+}
+
+/// Load a cross-service [`Config`](aws_types::config::Config) from the environment
+///
+/// This builder supports overriding individual components of the generated config. Overriding a component
+/// will prevent the standard resolution chain from occuring (eg. overriding a region will prevent an HTTP request
+/// to IMDS).
+#[derive(Default, Debug)]
+pub struct EnvLoader {
```

Comment:
It kind of seems like this should get its own module, or some config loader module, as it is separate from the providers.